### PR TITLE
feat(cli): '--lsp' flag to start a basic LSP server

### LINKED ISSUE #263: LSP Support?
Integrating this directly into an IDE in a standard way would probably help adoption.

### PR DESCRIPTION
**Pull request by tekumara**
Adds the `--lsp` flag which starts a basic LSP server running on stdin/out.

The initial capabilities of the server are minimal. It only provides diagnostics to identify typos. 

Future PRs will include code actions (ie: quick fixes) and support for config files. 

Resolves #263

### CODE REVIEW

**Comment by tekumara** [comment]
This is my first real contribution in Rust, so I'm open to suggestions on best practice here.

File: crates/typos-cli/src/bin/typos-cli/args.rs [hunk 1]
```diff
@@ -90,6 +90,10 @@ pub(crate) struct Args {
     #[arg(long, group = "mode")]
     pub(crate) type_list: bool,
 
+    /// Start the LSP server.
+    #[arg(long, group = "mode")]
+    pub(crate) lsp: bool,
```
[the remark this answers is not in the document]
Comment (tekumara) [reply crates/typos-cli/src/bin/typos-cli/args.rs:95]: No problem, and thanks for taking a look at this. At this initial point, the support offered would realistically just be me. Given that, would a `typos-lsp` crate within this repo provide enough of a distinction, or better to leave this out of this repo altogether? If the later, I'll probably end up moving this into [tekumara/typos-vscode](https://github.com/tekumara/typos-vscode) and may suggest small successive PRs as needed to make the integration work 🙏  I'll happily work with either option.

[the remark this answers is not in the document]
Comment (epage) [reply crates/typos-cli/src/bin/typos-cli/args.rs:95]: Thank you for being understanding!

Let's start outside the repo so I can better observe how all of this works before integrating it into my processes.

**Comment by mawkler** [comment]
@tekumara That's awesome that you're making an LSP out of typos! I use typos for spell checking inside of Neovim and wanted a code-action for correcting a word and for ignoring a spelling error, and found this issue.

It looks like typos-vscode is heavily coupled to VSC\*\*e. To add the language server to Neovim I really just need the `typos-lsp` binary. What would be the best way of extracting `typos-lsp`? Is it available as a crate? 🙂

**Pull request closed without merging**